src/main/java/com/exasol/adapter/dynamodb/documentfetcher/dynamodb/DynamodbQueryDocumentFetcher.java

```diff
@@ -3,32 +3,176 @@
      *
-     * @param queryRequest DynamoDB request
+     * @param genericParameters      generic parameters (not query specific)
+     * @param indexName              name of the index to scan
+     * @param keyConditionExpression condition expression for key columns
      */
-    DynamodbQueryDocumentFetcher(final QueryRequest queryRequest) {
-        this.queryRequest = queryRequest;
+    private DynamodbQueryDocumentFetcher(final GenericTableAccessParameters genericParameters, final String indexName,
+            final String keyConditionExpression) {
+        this.genericParameters = genericParameters;
+        this.indexName = indexName;
+        this.keyConditionExpression = keyConditionExpression;
+    }
+
+    static Builder builder() {
+        return new Builder();
     }
 
     QueryRequest getQueryRequest() {
-        return this.queryRequest;
+        final QueryRequest.Builder templateQueryRequest = QueryRequest.builder();
+        templateQueryRequest.tableName(this.genericParameters.getTableName());
+        if (!this.genericParameters.getExpressionAttributeNames().isEmpty()) {
+            templateQueryRequest.expressionAttributeNames(this.genericParameters.getExpressionAttributeNames());
+        }
+        if (!this.genericParameters.getFilterExpression().isEmpty()) {
+            templateQueryRequest.filterExpression(this.genericParameters.getFilterExpression());
+        }
+        if (!this.genericParameters.getExpressionAttributeValues().isEmpty()) {
+            templateQueryRequest.expressionAttributeValues(this.genericParameters.getExpressionAttributeValues());
+        }
+        if (!this.genericParameters.getProjectionExpression().isEmpty()) {
+            templateQueryRequest.projectionExpression(this.genericParameters.getProjectionExpression());
+        }
+        if (this.indexName != null) {
+            templateQueryRequest.indexName(this.indexName);
+        }
+        templateQueryRequest.keyConditionExpression(this.keyConditionExpression);
+        return templateQueryRequest.build();
     }
 
     @Override
-    public Stream<Map<String, AttributeValue>> run(final AmazonDynamoDB client) {
-        return client.query(this.queryRequest).getItems().stream();
+    public Stream<Map<String, AttributeValue>> run(final DynamoDbClient client) {
+        return client.query(this.getQueryRequest()).items().stream();
+    }
+
+    /**
+     * Builder for {@link DynamodbQueryDocumentFetcher}.
+     */
+    public static final class Builder {
+        private String tableName;
+        private Map<String, String> expressionAttributeNames;
+        private Map<String, DocumentValue<DynamodbNodeVisitor>> expressionAttributeValues;
+        private String filterExpression;
+        private String projectionExpression;
+        private String indexName;
+        private String keyConditionExpression;
+
+        private Builder() {
+        }
+
+        public static Builder aDynamodbScanDocumentFetcher() {
```

Comment:
What's this 'a'? in the method name. Should be a verb

Reply:
I did oversee this method. It was autogenerated...